scipy/signal/filter_design.py

```diff
@@ -162,9 +162,13 @@ def freqs(b, a, worN=None, plot=None):
     """
     if worN is None:
         w = findfreqs(b, a, 200)
-    elif isinstance(worN, int):
-        N = worN
-        w = findfreqs(b, a, N)
+    elif np.ndim(worN) == 0:
+        try:
+            N = operator.index(worN)
+        except TypeError:
+            pass
+        else:
+            w = findfreqs(b, a, N)
```

Comment:
You could save on branching/lines a little bit with `else` instead of `elif` here:
```
else:
    try:
        N = operator.index(worN)
    except TypeError:
        w = worN
    else:
        w = findfreqs(b, a, N)
```

Reply:
It needs to do the ndim check to interpret `array([8])` as "measure at point 8", rather than "measure at 8 points" since apparently older numpy versions returned .index() on 1-element N-D arrays.

Reply:
Using, for example, `worN=3.0`, results in `UnboundLocalError: local variable 'w' referenced before assignment`

Reply:
I modified it to fix the unbound error. Now it calls index twice, but that's not a big deal? Otherwise I'd have to duplicate other things and it would be harder to read. Still need to write tests for this condition

Reply:
Calling `index` twice seems fine if it leads to the simplest code.